x/bank/keeper/proposal_handler.go

```diff
@@ -0,0 +1,21 @@
+package keeper
+
+import (
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// HandleUpdateDenomMetadataProposal handles update of fee denom metadata
+func HandleUpdateDenomMetadataProposal(ctx sdk.Context, k Keeper, proposal *types.UpdateDenomMetadataProposal, registry codectypes.InterfaceRegistry) error {
```

Comment:
I think this is the old way of processing gov proposals. The new way involves adding an `authority` to the keeper (if it doesn't have one already). Then have an endpoint that processes the `UpdateDenomMetadataProposal` (e.g. in `tx.proto` and `msg_server.go`) restricting usage to just the keeper's authority.

From `simapp/app.go`:
```
// Register the proposal types
// Deprecated: Avoid adding new handlers, instead use the new proposal flow
// by granting the governance module the right to execute the message.
// See: https://github.com/cosmos/cosmos-sdk/blob/release/v0.46.x/x/gov/spec/01_concepts.md#proposal-messages
```
https://github.com/cosmos/cosmos-sdk/blob/release/v0.46.x/x/gov/spec/01_concepts.md#proposal-messages